pkg/debounce/debounce.go

```diff
@@ -34,8 +36,12 @@ func (d *Debouncer) debounceRoutine(ctx context.Context) {
 
 		err := d.function(ctx)
 		if err == nil {
-			// Reset scheduled to 0
-			atomic.StoreInt32(&d.scheduled, 0)
+			if d.calledDuringExecution.CompareAndSwap(true, false) {
+				d.debounceRoutine(ctx)
+			} else {
+				// Reset scheduled to 0
```

Comment:
leftover I guess

Reply:
yeah, better comments now because the use of atomic might not be intuitive in the beginning